.github/workflows/process_new_pull_request.yml

```diff
@@ -0,0 +1,230 @@
+              return allowed.test(input) ? input : '';
+            };
+            
+            const sanitizePrNumber = (num) => {
+              const n = Number(num);
+              return Number.isInteger(n) && n > 0 ? n : NaN;
+            };
+            
+            // Core PR metadata with validation
+            if (!context.payload?.pull_request || !context.payload.repository) {
+              core.setFailed('Invalid event payload');
+              return;
+            }
+            
+            const prNumber = sanitizePrNumber(context.payload.pull_request.number);
+            if (isNaN(prNumber)) {
+              core.setFailed(`Invalid PR number: ${context.payload.pull_request.number || 'undefined'}`);
+              return;
+            }
+            
+            const repo = sanitizeRepo(context.payload.repository.name);
+            const owner = sanitizeUsername(context.payload.repository.owner.login);
+            if (!repo || !owner) {
+              core.setFailed(`Invalid repo (${repo}) or owner (${owner})`);
+              return;
+            }
+            
+            // Sanitize PR author
+            const prAuthor = sanitizeUsername(context.payload.pull_request.user?.login);
+            if (!prAuthor) {
+              console.log('Unable to determine PR author');
+            }
+
+            // ======================
+            // CONTRIBUTOR CHECK
+            // ======================
+            const isNewContributor = await (async () => {
+              if (!prAuthor) return false;
+              
+              try {
+                const searchResults = await github.rest.search.issuesAndPullRequests({
+                  q: `repo:${owner}/${repo} type:pr is:merged author:${prAuthor}`
+                });
+                
+                return searchResults.data.total_count === 0;
+              } catch (error) {
+                // Fail noisily for unexpected errors
+                core.setFailed(`Contributor check failed: ${error.message}`);
+                throw error;
+              }
+            })();
+
+            // ======================
+            // REVIEWER VALIDATION
+            // ======================
+            const requestedIndividualReviewers = 
+              (context.payload.pull_request.requested_reviewers || [])
+                .map(r => sanitizeUsername(r?.login))
+                .filter(username => username);
+
+            const requestedTeamReviewers = 
+              (context.payload.pull_request.requested_teams || [])
+                .map(t => t?.slug)
+                .filter(slug => {
+                  // Allow single-character team names
+                  if (typeof slug !== 'string') return false;
+                  return /^[a-z\d](?:[a-z\d]|-[a-z\d])*$/i.test(slug);
+                });
+
+            const hasIndividualReviewers = requestedIndividualReviewers.length > 0;
+            const hasTeamReviewers = requestedTeamReviewers.length > 0;
```

Comment:
I don't think we need to check for team reviewers at all now that we're notifying whenever there are no individual reviewers. It doesn't look like we do anything with the team reviewers list now, except in the message to the dev workflow team. In that case, the problem we want to tell them about is that no individual reviewer is assigned. If a team but no individual is assigned, that will be obvious when dev workflow investigates, so there's no need to maintain extra code just to specify that in the notification message